tests/test_client.py

```diff
@@ -206,8 +270,10 @@ def test_helpful_error_message_received_on_connection_reset_error():
     """
     ws_client, _, _ = default_ws_client_setup("wss://this-url-wont-be-used:1")
 
-    async def mock_connect(*args, **kwargs):
+    @contextlib.asynccontextmanager
+    async def mock_connect(*_, **__):
         raise ConnectionResetError("foo")
+        yield None  # pylint: disable=unreachable
```

Comment:
Why do we have this yield? It does indeed look unreachable, if there is some other need for it, can you explain it in a comment?

Reply:
It's necessary to make the context manager work - I'll add a comment yeah.